rotkehlchen/exchanges/kraken.py

```diff
@@ -782,3 +779,282 @@ def query_online_income_loss_expense(
+        if pair not in self.pairs:
+            # Try tansaliting one and not the other
+            assets_symbols = [
+                WORLD_TO_KRAKEN.get(assets[0].symbol, assets[0].symbol),
+                assets[1].symbol,
+            ]
+            pair = "".join(assets_symbols)
+            if pair not in self.pairs:
+                pair = "".join(assets_symbols[::-1])
+            if pair not in self.pairs:
+                # Try the other way around
+                assets_symbols = [
+                    WORLD_TO_KRAKEN.get(assets[1].symbol, assets[1].symbol),
+                    assets[0].symbol,
+                ]
+                pair = "".join(assets_symbols)
+                if pair not in self.pairs:
+                    pair = "".join(assets_symbols[::-1])
+                if pair not in self.pairs:
+                    # Try translation both
+                    assets_symbols = [
+                        WORLD_TO_KRAKEN.get(asset.symbol, asset.symbol)
+                        for asset in assets
+                    ]
+                    pair = "".join(assets_symbols)
+                    if pair not in self.pairs:
+                        pair = "".join(assets_symbols[::-1])
+                    if pair not in self.pairs:
+                        return None
+        return pair
+
+    def process_kraken_trades(
+        self,
+        raw_data: List[HistoryEvent],
+    ) -> List[Trade]:
+        try:
+            if len(self.pairs) == 0:
+                self.query_pairs()
+        except RemoteError as e:
+            self.msg_aggregator.add_error(f'Failed to query kraken pairs due to {str(e)}')
+            raise RemoteError from e
+        trades = []
+        get_attr = operator.attrgetter('event_identifier')
+        grouped_events = [list(g) for k, g in itertools.groupby(sorted(raw_data, key=get_attr), get_attr)]  # noqa: E501
+        for trade_parts in grouped_events:
+            is_spend_receive = any(
+                [trade.event_type == HistoryEventType.RECEIVE for trade in trade_parts],
+            )
+            if is_spend_receive and len(trade_parts) < 2:
+                continue
+            trade_assets = [
+                trade.amount.asset
+                for trade in trade_parts
+                if (
+                    trade.amount.balance.amount != ZERO and
+                    trade.event_subtype != HistoryEventSubType.FEE
+                )
+            ]
+            pair = self.kraken_asset_to_pair(trade_assets)
+            if pair is None:
+                self.msg_aggregator.add_error(
+                    f'Couldnt find pair for assets {trade_assets} and '
+                    f'trade id {trade_parts[0].event_identifier}',
+                )
+                continue
+            base_asset, quote_asset = kraken_to_world_pair(pair)
+            timestamp = trade_parts[0].timestamp
+            base_trade, quote_trade, fee_part, kfee_part = None, None, None, None
```

Comment:
Yes, exactly is doing what you said. Well if there are more parts then something is off as we don't know how to process it or it would mean that kraken has changed how it works. afaik kraken only allows you to trade in the pair markets so there is no intermediate trade ever. This would be different in a on chain AMM